rest/config.go

```diff
@@ -331,7 +334,73 @@ func (dbConfig *DbConfig) setup(name string) error {
 		dbConfig.Server = &urlStr
 	}
 
-	return err
+	// If the sync function is referenced from an external file or http(s) endpoint
+	// unlike the inline JavaScript definition, read it from the specified path.
+	if dbConfig.Sync != nil {
+		sync, err := loadJavaScript(*dbConfig.Sync)
+		if err != nil {
+			return err
+		}
+		dbConfig.Sync = &sync
+	}
+
+	// If the import filter function is referenced from an external file or http(s) endpoint
+	// unlike the inline JavaScript definition, read it from the specified path.
+	if dbConfig.ImportFilter != nil {
+		importFilter, err := loadJavaScript(*dbConfig.ImportFilter)
+		if err != nil {
+			return err
+		}
+		dbConfig.ImportFilter = &importFilter
+	}
+
+	// If the conflict resolution function is referenced from an external file or http(s) endpoint
+	// unlike the inline JavaScript definition, read it from the specified path.
+	for _, rc := range dbConfig.Replications {
+		if rc.ConflictResolutionFn != "" {
+			conflictResolutionFn, err := loadJavaScript(rc.ConflictResolutionFn)
+			if err != nil {
+				return err
+			}
+			rc.ConflictResolutionFn = conflictResolutionFn
+		}
+	}
+
+	return nil
+}
+
+// loadJavaScript loads the JavaScript source from an external file or and HTTP/HTTPS endpoint.
+// If the specified path does not qualify for a valid file or an URI, it returns the input path
+// as-is with the assumption that it is an inline JavaScript source. Returns error if there is
+// any failure in reading the JavaScript file or URI.
+func loadJavaScript(path string) (js string, err error) {
```

Comment:
This looks like it clones a lot of code we already have in LoadServerConfig.  Did you evaluate whether the common functionality could be shared/reused?

Reply:
This was intentional. The main reason that I didn’t refactor the duplicate code in both LoadServerConfig and loadJavaScript in the first place was that there is no cleaner way to identify whether the given path is an inline JavaScript or a valid file. Aside from that I thought it would be easier to log the details while read file from URL or a path inside loadJavaScript. But I did refactor it in the latest commit.